Update slash command limits (#2667)
hammerandchisel/discord-api-docs: docs/interactions/Slash_Commands.md

In this documentation you'll find some notes about limits and caps on certain parts of Slash Commands. At a high level, they are as follows:

- An app can have up to 100 top-level global commands with unique names
- An app can have up to an additional 100 guild commands per guild
- An app can have up to 25 subcommand groups on a top-level command
- An app can have up to 25 subcommands within a subcommand group
- commands can have up to 25 `options`
- options can have up to 25 `choices`

> - Maximum of 4000 characters for combined name, description, and value properties for each command and its children

Reviewer: So I am encouraged to try having a shorter description for each option?

Additionally, what does "for each command and its children" mean? What is an example of a command invocation that *wouldn't* work?

> - Maximum of 4000 characters for combined name, description, and value properties for each command and its children

Reviewer 2: > description, and value properties for each command and its subcommands and groups

- Limitations on [command names](#DOCS_INTERACTIONS_SLASH_COMMANDS/registering-a-command)
- Limitations on [nesting subcommands and groups](#DOCS_INTERACTIONS_SLASH_COMMANDS/nested-subcommands-and-groups)

> - Rate limitation of 2 requests per 60 seconds for batch update

Reviewer: Is the batch update for guilds keyed by guild ID? (Yes I know guild ID is a major parameter in this route, I just have to make sure though :')

```suggestion
- Rate limit of 2 requests per 60 seconds for batch update
```

> - Rate limitation of 5 requests per 20 seconds for all other routes

Reviewer: ```suggestion
- Rate limit of 5 requests per 20 seconds for all other routes
```

Reviewer 3: Only `create`, `edit` and `delete` routes have this limit, `get`, `get-all` ones  are unlimited.

> - Global rate limitation of 200 application command creates per day per guild

Reviewer: Does the batch update factor into this?

If so, will it just add as many commands as it can before failing? Or will it fail early? Or will you lose the rest of your rate limit and none of the commands be registered?

```suggestion
- Global rate limit of 200 application command creates per day per guild
```

If you specify `choices` for an option, they are the **only** valid values for a user to pick

| Field | Type          | Description                                     |
|-------|---------------|-------------------------------------------------|
| name  | string        | 1-100 character choice name                     |

> | value | string or int | value of the choice, 0-100 characters if string |

Reviewer: ```suggestion
| value | string or int | value of the choice, up to 100 characters if string |
```
